### PR TITLE
 JUnit 5 Dynamic tests not showing with their displayName #5975 

### PR DESCRIPTION
**Pull request by uberto**
Fixes:  JUnit 5 Dynamic tests not showing with their displayName #5975 

### Context

This is a fix for the issue:
https://github.com/gradle/gradle/issues/5975

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy [hunk 1]
```diff
@@ -39,6 +41,30 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
             '''
     }
 
+    void createDynamicJupiterTest() {
```
Comment (6hundreds) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy:44]: You are testing logging, right? If so, there is more appropriate place for such tests IMO - JUnitPlatformLoggingIntegrationTest. Consider to move your tests there

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy [hunk 2]
```diff
@@ -52,6 +78,84 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
         succeeds('test')
     }
 
+    def 'can handle dynamically created test'() {
+        given:
+        buildFile << """
+        dependencies {
+            testImplementation 'org.junit.platform:junit-platform-runner:1.0.3'
```
Comment (6hundreds) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy:85]: Do you need this line at all?
At glance JUnitPlatformIntegrationSpec doing all the configuration that you need.

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy [hunk 3]
```diff
@@ -52,6 +78,84 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
...
+        }
+        """
+        createDynamicJupiterTest()
+
+        when:
```
Comment (6hundreds) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy:90]: This when/then setup looking more complicated than need. Probably same way as JUnitPlatformLoggingIntegrationTest doing will allow you to achieve same behavior more ideomatically?

File: subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy [hunk 4]
```diff
@@ -52,6 +78,84 @@ class JUnitPlatformIntegrationTest extends JUnitPlatformIntegrationSpec {
...
+        new DefaultTestExecutionResult(testDirectory)
+            .assertTestClassesExecuted('org.gradle.JUnitJupiterDynamicTest')
+            .testClass('org.gradle.JUnitJupiterDynamicTest')
+            .assertTestCount(3, 1, 0)
+
+        where:
+        console << [ConsoleOutput.Plain] //, ConsoleOutput.Rich, ConsoleOutput.Verbose
+    }
+
+    def 'can handle dynamically created test on log info'() {
```
Comment (6hundreds) [inline subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/junitplatform/JUnitPlatformIntegrationTest.groovy:118]: Both tests looks same excluding passing --info option. I'm not sure that both tests are needed (we could discuss) but if it is then we could reduce this duplication with Spock's data pipes

File: subprojects/core/src/main/java/org/gradle/testfixtures/internal/TestBuildRootBuildOperationType.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testfixtures.internal;
+
+import org.gradle.internal.operations.BuildOperationType;
+
+public interface TestBuildRootBuildOperationType extends BuildOperationType<TestBuildRootBuildOperationType.Details, TestBuildRootBuildOperationType.Result> {
```
Comment (6hundreds) [inline subprojects/core/src/main/java/org/gradle/testfixtures/internal/TestBuildRootBuildOperationType.java:21]: Seems like some of changes are not related to your PR, right?
Could you fix it? Let me know if you'll have a problems with it

**Comment by uberto** [comment]
I've spent something like 10 hours on this. 2 hours on the fix and the rest
in making sure the pr is accepted, and failing at it.
There shouldn't be any changes not related. Could be a merge artifact tho.

As soon as I have some other time, I'll try again applying the 3/4 file
changes from a clean fork.

Uberto

**Comment by 6hundreds** [comment]
@uberto Hey, unintentionally closed PR - was tried to fix git state here. Sorry for that. Would you open new one from clean fork?

**Comment by uberto** [comment]
Yes I will do. Hopefully this weekend.

Uberto

**Comment by robd** [comment]
@6hundreds I am keen to fix this, but wanted to understand requirements to get this merged. From above seems like maybe not keen to merge / fix, so wanted to check.

**Comment by 6hundreds** [comment]
@robd Hi Rob! Thank you for intention to make contribution to Gradle, we are really appreciate it.

About requirements you can read in Gradle contribution guide https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md

This PR was closed accidentally by me because of broken git state into it. So new PR is welcome
